sympy/matrices/matrices.py

```diff
@@ -4053,34 +4053,49 @@ def QRdecomposition(self):
         cls = self.__class__
         mat = self.as_mutable()
 
-        if not mat.rows >= mat.cols:
-            raise MatrixError(
-                "The number of rows must be greater than columns")
         n = mat.rows
         m = mat.cols
-        rank = n
-        row_reduced = mat.rref()[0]
-        for i in range(row_reduced.rows):
-            if row_reduced.row(i).norm() == 0:
-                rank -= 1
-        if not rank == mat.cols:
-            raise MatrixError("The rank of the matrix must match the columns")
+        ranked = list()
+
+        # If matrix is fat, pad additional zeros to make it square.
```

Comment:
suggest:

# pad with additional rows to make wide matrices square;
# nOrig keeps track of original size so zeros can be trimmed from Q